### PR TITLE
Enable Intel Ethernet Connection E800 networking driver

### PR DESCRIPTION
**Pull request by christopherco**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./LICENSES-AND-NOTICES/SPECS/data/licenses.json`, `./LICENSES-AND-NOTICES/SPECS/LICENSES-MAP.md`, `./LICENSES-AND-NOTICES/SPECS/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

This change enables the Intel Ethernet Connection E800 series of networking drivers as loadable modules in our Azure Linux 3.0 kernels.

Signed-off-by: Chris Co <chrco@microsoft.com>

###### Does this affect the toolchain?  

**YES**

###### Associated issues  

- https://microsoft.visualstudio.com/OS/_workitems/edit/54512964

###### Test Methodology

- Pipeline build id: [670789](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=670789&view=results)

### CODE REVIEW

File: SPECS/kernel/config [hunk 1]
```diff
@@ -2691,7 +2691,9 @@ CONFIG_I40E=m
 CONFIG_I40E_DCB=y
 CONFIG_IAVF=m
 CONFIG_I40EVF=m
-# CONFIG_ICE is not set
+CONFIG_ICE=m
```
Comment (rlmenge) [inline SPECS/kernel/config:2694]: Kconfigs
[CONFIG_ICE](https://github.com/microsoft/CBL-Mariner-Linux-Kernel/blob/rolling-lts/mariner-3/6.6.57.1/drivers/net/ethernet/intel/Kconfig#L277)
[CONFIG_ICE_SWITCHDEV](https://github.com/microsoft/CBL-Mariner-Linux-Kernel/blob/rolling-lts/mariner-3/6.6.57.1/drivers/net/ethernet/intel/Kconfig#L300)
[CONFIG_ICE_HWTS](https://github.com/microsoft/CBL-Mariner-Linux-Kernel/blob/rolling-lts/mariner-3/6.6.57.1/drivers/net/ethernet/intel/Kconfig#L314)
[CONFIG_INFINIBAND_IRDMA](https://github.com/microsoft/CBL-Mariner-Linux-Kernel/blob/rolling-lts/mariner-3/6.6.57.1/drivers/infiniband/hw/irdma/Kconfig)
[CONFIG_PLDMFW ](https://github.com/microsoft/CBL-Mariner-Linux-Kernel/blob/rolling-lts/mariner-3/6.6.57.1/lib/Kconfig#L752)

**Review by rlmenge — APPROVED** [review]
Toolchain builds 
[AMD](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=673494&view=results)
[ARM ](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=673495&view=results)

[Buddy Build
](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=673474&view=results)

image

**Comment by christopherco** [comment]
Updated buddy build - https://dev.azure.com/mariner-org/mariner/_build/results?buildId=683381&view=results

**Comment by christopherco** [comment]
> Updated buddy build - https://dev.azure.com/mariner-org/mariner/_build/results?buildId=683381&view=results

ICE kernel module present on both architectures with most recent buddy build packages.

image

image

**Review by eric-desrochers — APPROVED** [review]
Approving on behalf of the Stable Maintainer Team and per discussion with Chris Co.

**Pull request merged**